lib/errors/index.ts

```diff
@@ -39,13 +39,18 @@ export class ArsenalError extends Error {
     /** Object used to determine the error type.
      * Example: error.is.InternalError */
     #is: Is;
+    /** A map of error metadata (can be extra fields
+     * that only show in debug mode) */
+    #metadata: Map<string, Array<Object>>;
```

Comment:
I usually try to write arrays as `Object[]` instead of `Array<Object>`, in case where type can be inlined. (I don't do it for types spanning accros multiple lines.)
It's mainly a matter of style, and would be good to have an opinion about that to avoid commits changing style later. @jbertran @Kerkesni any opinion?

Reply:
To me, `Type[]` seems less verbose, but still quite obvious, so I'm in favour of pushing in that direction

Reply:
yeah i agree, it's cleaner that way 🙂